File: admin-service/src/main/java/com/shopflow/admin/service/AdminService.java
package com.shopflow.admin.service;

import com.shopflow.admin.dto.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.ZonedDateTime;
import java.util.List;
import java.util.UUID;

public interface AdminService {

    // User management
    Page<UserResponse> getAllUsers(Pageable pageable);
    void banUser(UUID adminId, UUID userId);
    void unbanUser(UUID adminId, UUID userId);

    // Seller management
    Page<SellerResponse> getAllSellers(Pageable pageable);
    List<SellerResponse> getPendingSellers();
    void approveSeller(UUID adminId, UUID sellerId);
    void rejectSeller(UUID adminId, UUID sellerId);
    void updateCommission(UUID adminId, UUID sellerId, UpdateCommissionRequest request);

    // Product management
    Page<ProductResponse> getAllProducts(Pageable pageable);
    void deleteProduct(UUID adminId, UUID productId);

    // Order + Return management
    Page<OrderResponse> getAllOrders(Pageable pageable);
    Page<ReturnRequestResponse> getAllReturnRequests(Pageable pageable);
    void approveReturn(UUID adminId, UUID returnId, ApproveReturnRequest request);
    void rejectReturn(UUID adminId, UUID returnId, ApproveReturnRequest request);

    // Reports
    RevenueReportResponse getRevenueReport(ZonedDateTime from, ZonedDateTime to);

    // Audit
    Page<AuditLogResponse> getAuditLogs(UUID adminId, Pageable pageable);
}

File: shopflow-frontend/src/services/modules/admin.service.js
import api from '../api/axiosInstance';

export const adminService = {
  // Users
  getAllUsers: async (page = 0, size = 20) => {
    const response = await api.get(`/admin/users?page=${page}&size=${size}`);
    return response.data;
  },
  
  banUser: async (userId) => {
    return await api.put(`/admin/users/${userId}/ban`);
  },
  
  unbanUser: async (userId) => {
    return await api.put(`/admin/users/${userId}/unban`);
  },
  
  // Products
  getAllProducts: async (page = 0, size = 20) => {
    const response = await api.get(`/admin/products?page=${page}&size=${size}`);
    return response.data;
  },
  
  deleteProduct: async (productId) => {
    return await api.delete(`/admin/products/${productId}`);
  },
  
  // Orders
  getAllOrders: async (page = 0, size = 20) => {
    const response = await api.get(`/admin/orders?page=${page}&size=${size}`);
    return response.data;
  },
  
  // Reports
  getSalesReport: async (startDate, endDate) => {
    const response = await api.get(`/admin/reports/sales?start=${startDate}&end=${endDate}`);
    return response.data;
  },

  // ─── Seller Management ──────────────────────────────────────────────
  getPendingSellers: async () => {
    const response = await api.get('/admin/sellers/pending');
    return response.data;
  },

  approveSeller: async (id) => {
    return await api.put(`/admin/sellers/${id}/approve`);
  },

  rejectSeller: async (id) => {
    return await api.put(`/admin/sellers/${id}/reject`);
  },

  updateCommission: async (id, rate) => {
    return await api.put(`/admin/sellers/${id}/commission`, { commissionRate: rate });
  }
};
